### PR TITLE
Convert `canvas-renderer` to typescript

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
A hour ago I wanted to make a draft. 

I woke up when it was done.

Notice how `GlobalMixins` helped in that case.

I moved out `_canvasRenderTarget` field from `BaseRenderTexture` , we can move class `CanvasRenderTarget` from `PIXI.utils` in another PR.

Another idea is that we can move `forcePixels` to globalMixins too, thus preventing people from passing it to `autoDetect` if they didnt use `pixi.js-legacy` , - also for discussion in separate PR.

### CODE REVIEW

File: packages/canvas/canvas-renderer/src/CanvasRenderer.ts [hunk 1]
```diff
@@ -38,16 +68,16 @@ export class CanvasRenderer extends AbstractRenderer
...
-        super(RENDERER_TYPE.CANVAS, options, arg2, arg3);
+        super(RENDERER_TYPE.CANVAS, options);
 
         /**
          * The root canvas 2d context that everything is drawn with.
          *
          * @member {CanvasRenderingContext2D}
          */
-        this.rootContext = this.view.getContext('2d', { alpha: this.transparent });
+        this.rootContext = this.view.getContext('2d', { alpha: this.transparent }) as any;
```
Comment (bigtimebuddy) [inline packages/canvas/canvas-renderer/src/CanvasRenderer.ts]: Why `as any`?

**Comment by Zyie** [comment]
This looks good to me as well. 

Can you add `public` to the class methods? I assume these all are not meant to be `internal`

File: packages/canvas/canvas-renderer/src/CanvasRenderer.ts (hunk 1 above)
Comment (ivanpopelyshev) [reply packages/canvas/canvas-renderer/src/CanvasRenderer.ts]: Because line is too long

Comment (bigtimebuddy) [reply packages/canvas/canvas-renderer/src/CanvasRenderer.ts]: So break it into two. Using `any` should be a last-resort for resolving typing issues, not workaround linting

**Pull request merged**